pkg/api/message/v1/service.go

```diff
@@ -367,6 +370,24 @@ func (s *Service) Query(ctx context.Context, req *proto.QueryRequest) (*proto.Qu
 		}
 	}
 
+	var hasV3Topic, hasNonV3Topic bool
+	for _, contentTopic := range req.ContentTopics {
+		if topic.IsV3(contentTopic) {
+			hasV3Topic = true
+		} else {
+			hasNonV3Topic = true
+		}
+	}
+	if hasV3Topic && hasNonV3Topic {
+		return nil, errors.New("cannot query v3 topics with non-v3 topics")
+	}
+	if hasV3Topic {
+		if s.mlsStore == nil {
+			return nil, errors.New("mls store not configured")
+		}
+		return s.mlsStore.QueryMessages(ctx, req)
```

Comment:
Alternatively we could add `QueryMessages` to the mls/v3 service with it's own isolated types, rather than hijacking the messagev1 service+types for it

Reply:
Downstream impact isn't a huge deal right now. There isn't that much downstream at the moment. Basically a single caller.

Reply:
Ok great I think I'll add the mls/v3 protos to separate this all out cleanly then